cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -481,6 +482,31 @@ protected void extract(Path file, Path targetDir) {
       } else {
         throw new IllegalStateException("Unknown archive format " + extension + ". Can not extract " + file);
       }
+
+      // collapse single parent directories
+      File currentDir = new File(targetDir.toString());
+      String[] currentSubFiles = currentDir.list();
+      if (currentSubFiles == null) {
+        throw new IllegalStateException("Extracted file does not contain any file or directories");
+      }
+      // TODO add wbin as well? (in case of msi azure installation)
```

Comment:
Not needed here `az` has a `CLI2` folder containing various files and folders including `wbin`.
These exclusions listed here are to avoid that we omit a folder that is required.
E.g. if your archive only contains
* bin
   * tool.exe
   * tool.dll

Then we still want to keep the `bin` folder in the `targetDir`.